Implementado layout inicial da pagina home e o unit tests

joaojvds opened the pull request:
--Colocado as cores do site em variáveis
--Separado mensagem de erro em componente
--Criado arquivos das paginas main e not found
--Configurado rotas no arquivo router
--alterado aparência do header
--alterado aparência das paginas de login e cadastro
--consertado redirect da pagina login
--criado estilo inicial da pagina home
--O navbar agora se adapta a qualquer tamanho de tela
--Adicionado um drawer menu para a navegação no mobile
--Adicionado unit tests no front end

joaojvds committed:
Criado variaveis CSS

--Colocado as cores do site em variaveis
--Separado mensagem de erro em componente

joaojvds committed:
Criado paginas main e not found

--Criado arquivos das paginas main e not found
--Configurado rotas no arquivo router

joaojvds committed:
Criado paginas main e not found

--Criado arquivos das paginas main e not found
--Configurado rotas no arquivo router

joaojvds committed:
Atualizado estilo da pagina

--alterado aparencia do header
--alterado aparencia das paginas de login e cadastro
--consertado redirect da pagina login
--criado estilo inicial da pagina home

joaojvds committed:
Adicionado navbar responsiva 

--O navbar agora se adapta a qualquer tamanho de tela
--Adicionado um drawer menu para a navegação no mobile

joaojvds committed:
unit test do frontend (#13)

--Adicionado testes de renderização em todos os arquivos
--Adicionado testes de passagem de dados entre os inputs e os estados nas paginas de login e cadastro
--Adicionado testes no carregamentos dos componentes em nas paginas de login e cadastro

joaojvds committed:
Merge branch 'master' into main-page

joaojvds committed:
Adicionado botão de logout

--Removido a propriedade do httpOnly do cookie enviado pelo servidor
--Botão logout só aparece quando a um cookie de sessão
--Adicionado função ao botão para apagar o cookie de sessão

GCastilho commented on client/src/index.js:
```diff
@@ -1,15 +1,17 @@
+                <Routes/>
+            </div>
+        </BrowserRouter>
     , document.getElementById('root'));
```
Esse arquivo não deveria se chamar index.jsx, com o X, já que é um arquivo do React?

GCastilho commented on client/src/pages/login/Login.jsx:
```diff
@@ -31,21 +32,24 @@ export default class Login extends React.Component {
+                <form className='login-form'>
                     <h1>Login</h1>
-
-                    {this.state.error ? <span className='login-error'>Senha ou usuario incorrentos!</span> : null}
```
Inverte essa mensagem para "Usuário ou senha incorretos", com o "usuário" primeiro

GCastilho commented on client/src/pages/login/Login.jsx:
```diff
@@ -31,21 +32,24 @@ export default class Login extends React.Component {
+                this.setState({errorMsg: 'Senha ou usuario incorrentos!'});
+            } else {
+                this.setState({errorMsg: 'Não foi possivel fazer o login'});
+            }
```
Isso vai dar conflito com o #16, só pra deixar anotado

GCastilho commented on client/src/pages/login/Login.jsx:
```diff
@@ -31,21 +32,24 @@ export default class Login extends React.Component {
             email: this.state.email,
             password: this.state.password
-        }).then(res => {
+        },{headers: {'X-Requested-With': 'XMLHttpRequest'},withCredentials: true}).then(res => {
```
> `{headers: {'X-Requested-With': 'XMLHttpRequest'},withCredentials: true}`

O que isso faz?

GCastilho commented on src/router/login.js:
```diff
@@ -66,7 +66,7 @@ Router.post('/', function(req, res) {
 		 * @todo cookie ter tempo de expiração
 		 */
-		res.cookie('sessionID', cookie.sessionID, { httpOnly:true })
+		res.cookie('sessionID', cookie.sessionID, { httpOnly:false })
```
Se ele vai ser false, o mais simples é remover o objeto de configuração inteiro, pois o resultado vai ser o mesmo

GCastilho commented on client/src/components/Header/Header.jsx:
```diff
@@ -1,13 +1,51 @@
+import Backdrop from "../Backdrop/Backdrop";
+
+export default () => {
+
```
Remove essa linha em branco

GCastilho commented on client/src/components/Header/Header.jsx:
```diff
@@ -1,13 +1,51 @@
+                            < Link to = "/register" className="button">Cadastro</Link>
+                        </> :
+                        <button className="button" onClick={logOutHandle}>Log out</button>
+                    }
```
Se é um `if` que checa uma negação, por que não inverte a ordem dos retornos eu checa se é truthy ao invés de chegar a negação do thuthy?

GCastilho commented on client/src/components/SideDrawer/SideDrawer.jsx:
```diff
@@ -0,0 +1,34 @@
+                        <li><Link to = "/register" className="drawer-button">Cadastro</Link></li>
+                    </>:
+                    <li><button className="drawer-button" onClick={logOutHandle}>Log out</button></li>
+                }
```
Mesmo comentário do outro `if` negado, pq não inverte?

GCastilho commented on client/src/index.js:
```diff
@@ -1,15 +1,17 @@
+import {BrowserRouter} from 'react-router-dom';
+
 import './index.css';
+
```
Algum motivo pra esse import estar entre dois blocos de import e separado por newLine em cima e em baixo?

GCastilho reviewed (changes requested):
-- Os testes não parecem estar funcionando, testei rodando todos e eles deram erro
-- As páginas de login e cadastro não são verticalmente responsivas, ou seja, se a janela não é muito alta, ela fica com uma barra de rolagem mínima na lateral, faça um jeito de ela se adaptar a resolução da dela, pelo menos em desktops (teste com com monitores com menor resolução)
-- A página também não é responsiva no mobile (já q vc quis resolver isso), ficando com fundo branco ao colocar dimensões de tela incomuns
-- Faça o botão de logout redirecionar para a HOME
-- E que tal adicionar uns **comentários** explicando o que cada parte faz e por que tá escrito do jeito que está escrito? Eu não vi nenhum comentário no seu código inteiro

joaojvds committed:
Adicionado comentarios nos arquivos

--Refatorado funções dos arquivos login, register e header
--Adicionado redirect para a home se estiver logado nas paginas login e register
--Botão logout faz o redirect para a home se apertado

joaojvds committed:
Consertado componente header

--Agora não tem que recarregar a pagina para adicionar o botão logout na navbar

joaojvds committed:
Refatorado SideDrawer.jsx

--Removido repetição de codigo
--Menu lateral fecha ao aperta um botão

joaojvds committed:
Alterado cores da navbar mobile

--Mundado aparencia da navbar vertical
--Consertado fundo branco ao colocar usar em resoluções pequenas

joaojvds committed:
Adicionado check na pagina login

--Antes de fazer o redirect, ele checa se o cookie foi recebido, se não foi ele da uma mensagem de erro
--Alterado cor do input

GCastilho commented on client/src/components/InputField/InputField.jsx:
```diff
@@ -1,7 +1,12 @@
 
 import './InputField.css';
 
+//Input com label flutuante
```
Comentários desse tipo (o de baixo) explicam o que o componente faz, certo? E o arquivo **é** o componente, correto? Nesse caso, coloca esse comentário no comentário header (o de cima), ex:

```
/*
 * client/src/components/InputField/InputField.jsx
 *
 * Input com label flutuante
 */
```

Faça isso em todos os comentários que se encaixam nessa descrição (não só nesse)

GCastilho commented on client/src/components/SideDrawer/DrawerToggleButton.jsx:
```diff
@@ -1,7 +1,12 @@
 
 import './DrawerToggleButton.css';
 
+//Botão com os 3 riscos verticais
```
>//Botão com os 3 **barras** verticais

GCastilho commented on client/src/components/SideDrawer/SideDrawer.jsx:
```diff
@@ -1,32 +1,55 @@
+    const [redirect, setRedirect] = React.useState(false);
 
+    //Handle do botão logout, ao aperta-lo ele apaga o cookie, ativa o botão de login e ativa o redirect
     const logOutHandle = () => {
```
Esse tipo de comentário, que explica o que um **bloco** faz, poderia ser feito usando a syntax do [jsdoc](https://devdocs.io/jsdoc/), que se resume em

```
/**
 * Descrição aqui
/*
```

GCastilho commented on client/src/pages/login/Login.jsx:
```diff
@@ -1,55 +1,85 @@
             errorMsg: ''
         };
+
+        //garanti o contexto quando estiver chamando a função com o this
```
garante é com E

GCastilho commented on client/src/pages/login/Login.jsx:
```diff
@@ -1,55 +1,85 @@
-        },{headers: {'X-Requested-With': 'XMLHttpRequest'},withCredentials: true}).then(res => {
+        }).then(res => {
+            //Se o post foi bem sucedido ele fara o redirect
+            // this.setState({redirect: true});
```
Pq esse bloco de código tá comentado?

GCastilho reviewed (changes requested):
Algumas mudanças em comentários; comentários referentes a um bloco de texto devem ser feitos usando jsdoc, comentários que se referem a o que o arquivo faz (como comentários que dizem o que um componente faz, q vc botou na exports) devem ser colocados no comentário do header (eu dei um exemplo, mas vale pra todos)

joaojvds committed:
Arrumado os comentarios

--Mudado os comentarios de bloco a pedido do GCastilho

joaojvds committed:
Arrumado um pouco mais os comentarios

--Colocado os comentarios de acordo com o jsdocs
